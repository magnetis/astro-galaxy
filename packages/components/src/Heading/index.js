import styled from 'styled-components';
import { color, space, layout, typography } from 'styled-system';

const Heading = styled.h1`
  ${color}
  ${space}
  ${layout}
  ${typography}
  font: ${props => props.theme.fonts.primary};
  font-size: ${props => props.theme.fontSizes.headings[props.fontSize]};
  ${props => props.fontSize === 'huge' && 'font-weight: 700'}
`;

Heading.displayName = 'Heading';

Heading.defaultProps = {
  color: 'moon900',
  fontSize: 'large',
};

Heading.propTypes = {
  ...color.propTypes,
  ...space.propTypes,
  ...layout.propTypes,
  ...typography.propTypes,
};

export default Heading;
