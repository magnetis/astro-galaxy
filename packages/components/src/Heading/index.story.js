import { storiesOf } from '@storybook/react';
import React from 'react';
import Heading from './index.js';

storiesOf('typography|heading', module)
  .add('default', () => <Heading>Astro Galaxy</Heading>)
  .add('colors', () => (
    <>
      <Heading color="earth400">earth400</Heading>
      <Heading color="mars400">mars400</Heading>
      <Heading color="moon400">moon400</Heading>
      <Heading color="space400">space400</Heading>
      <Heading color="sun400">sun400</Heading>
      <Heading color="uranus400">uranus400</Heading>
      <Heading color="venus400">venus400</Heading>
    </>
  ))
  .add('sizes', () => (
    <>
      <Heading fontSize="small">small</Heading>
      <Heading fontSize="medium">medium</Heading>
      <Heading fontSize="large">large</Heading>
      <Heading fontSize="huge">huge</Heading>
    </>
  ))
  .add('as <h{number}>', () => (
    <>
      <Heading>H1</Heading>
      <Heading as="h2">H2</Heading>
      <Heading as="h3">H3</Heading>
      <Heading as="h4">H4</Heading>
      <Heading as="h5">H5</Heading>
      <Heading as="h6">H6</Heading>
    </>
  ));
