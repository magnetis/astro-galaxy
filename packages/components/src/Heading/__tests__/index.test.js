import React from 'react';

import Heading from '../index';
import { theme } from '@magnetis/astro-galaxy-core';

describe('Heading', () => {
  it('renders correctly', () => {
    const json = rendererCreateWithTheme(<Heading>Heading</Heading>).toJSON();

    expect(json).toMatchSnapshot();
  });

  describe('when heading "color" prop is set', () => {
    it('renders with "mars600"', () => {
      const json = rendererCreateWithTheme(<Heading color="mars600">Heading</Heading>).toJSON();
      expect(json).toHaveStyleRule('color', theme.colors.mars600);
    });
  });

  describe('when heading "fontSize" prop is set', () => {
    it('renders with "small" font size ', () => {
      const json = rendererCreateWithTheme(<Heading fontSize="small">Heading</Heading>).toJSON();
      expect(json).toHaveStyleRule('font-size', theme.fontSizes.headings.small);
    });
  });
});
