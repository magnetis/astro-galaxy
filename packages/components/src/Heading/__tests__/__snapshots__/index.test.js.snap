// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading renders correctly 1`] = `
.c0 {
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  color: #32383c;
  font-size: large;
  font: normal normal 600 normal 16px/1.5 Poppins,sans-serif;
  font-size: 48px;
}

<h1
  className="c0"
  color="moon900"
  fontSize="large"
>
  Heading
</h1>
`;
